code/parachain/frame/ibc/src/lib.rs

```diff
@@ -406,9 +406,7 @@ pub mod pallet {
 	#[pallet::generate_deposit(pub (super) fn deposit_event)]
 	pub enum Event<T: Config> {
 		/// Events emitted by the ibc subsystem
-		Events { events: Vec<events::IbcEvent> },
-		/// Errors emitted by the ibc subsystem
-		Errors { errors: Vec<errors::IbcError> },
+		Events { events: Vec<Result<events::IbcEvent, errors::IbcError>> },
```

Comment:
You need to modify the `block_events` runtime_api implementation in the runtime file after this change.